examples/io-descriptors/README.md

```diff
@@ -0,0 +1,120 @@
+
+```yaml
+service: "service.py:ImageResize"
+include:
+  - "service.py"
+```
+
+After you finished configuring `bentofile.yaml`, run `bentoml serve .` to deploy the service locally. You can then interact with the auto-generated swagger UI to play
+around with each different API endpoints.
+
+## Different data types
+
+### Standard Python types
+
+The following demonstrates a simple addtion service, with both inputs and output as float parameters. You can
+obviously change the type annotation to `int`, `str` etc. to get familiar with the interaction between type
+annotaions and the auto-generated Swagger UI when deploying locally.\
+
+```python
+@bentoml.service()
+class AdditionService:
+
+    @bentoml.api()
+    def add(self, num1: float, num2: float) -> float:
+        return num1 + num2
+```
+
+### Files
+
+Files are handled through `pathlib.Path` in BentoML (which means you should handle the file as a file path in your API implementation as well as on the client side).
+Most file types can be specified through `bentoml.validators.Contentype(<file_type>)`. The input of this function follows the standard of the
+request format (such as `text/plain`, `application/pdf`, `audio/mp3` etc.).
+
+##### Appending Strings to File example
+```python
+@bentoml.service()
+class AppendStringToFile:
+
+    @bentoml.api()
+    def append_string_to_eof(
+        self,
+        txt_file: t.Annotated[Path, bentoml.validators.ContentType("text/plain")], input_string: str
+        ) -> t.Annotated[Path, bentoml.validators.ContentType("text/plain")]:
+        with open(txt_file, "a") as file:
+            file.write(input_string)
+        return txt_file
+```
+
+Within `service.py`, example API services with 4 different file types are implemented (audio, image, text file, and pdf file). The functionality of each service
+is quite simple and self-explainatory.
+
+Notice that for class `ImageResize`, two different API endpoints are implemented. This is because BentoML can support images parameters directly through
+`PIL.Image.Image`, which means that image objects can be directly passed through clients, instead of a file object.
+
+The last two services are examples of having `numpy.ndarray` or `pandas.DataFrame` as input parameters. Since they all work quite similarly with the above examples,
+we will not specifically explain them in this tutorial. You can try to write revise the service with `torch.Tensor` as input to check your understanding.
+
+To serve the these examples locally, run `bentoml serve .`
+
+```bash
+bentoml serve .
+2024-03-22T19:25:24+0000 [INFO] [cli] Starting production HTTP BentoServer from "service:ImageResize" listening on http://localhost:3000 (Press CTRL+C to quit)
+```
+
+Open your web browser at http://0.0.0.0:3000 to view the Swagger UI for sending test requests.
+
+You may also send request with `curl` command or any HTTP client, e.g.:
+
+```bash
+curl -X 'POST' \
+  'http://0.0.0.0:3000/classify_generated_texts' \
```

Comment:
Let's interact with one of the endpoints in this example. `classify_generated_texts` is not included in this example.